packages/integrations/mdx/src/vite-plugin-mdx.ts

```diff
@@ -16,6 +16,10 @@ export function vitePluginMdx(mdxOptions: MdxOptions): Plugin {
 			processor = undefined;
 		},
 		configResolved(resolved) {
+			// `mdxOptions` should be populated at this point, but `astro sync` doesn't call `astro:config:done` :(
+			// Workaround this for now by skipping here. `astro sync` shouldn't call the `transform()` hook here anyways.
```

Comment:
Interesting. @bholmesdev mentioned an issue with `astro sync` and Astro DB the other day, which I wonder if it might be related — it also uses `astro:config:done` in a similar pattern to this to assign settled integration state for use by plugins.

https://github.com/withastro/astro/blob/5bf5dd496bf55c0713754f2a4f7726693b759744/packages/db/src/core/integration/index.ts#L88-L89

Maybe something that will need fixing in `sync` itself?

Reply:
Yeah I think it would be nice if the hook was called in Astro sync too. I initially did that, but walked back a bit because I'm unsure if this breaks the db integration.

I noticed the db integration is running the same startup code in both `astro sync` and `astro:config:done`, and I'm unsure what happens in `astro sync`  if it runs that code twice. It should be something we can look into fixing in the future though.